test/src/main/java/org/apache/accumulo/test/DeleteCancelsMajCIT.java

```diff
@@ -0,0 +1,112 @@
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.apache.accumulo.core.client.Accumulo;
+import org.apache.accumulo.core.client.AccumuloClient;
+import org.apache.accumulo.core.client.BatchWriter;
+import org.apache.accumulo.core.client.IteratorSetting;
+import org.apache.accumulo.core.data.Mutation;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.IteratorUtil;
+import org.apache.accumulo.harness.SharedMiniClusterBase;
+import org.apache.accumulo.test.functional.SlowIterator;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DeleteCancelsMajCIT extends SharedMiniClusterBase {
+  private static final Logger log = LoggerFactory.getLogger(DeleteCancelsMajCIT.class);
+
+  AtomicBoolean started = new AtomicBoolean(false);
+  AtomicBoolean finished = new AtomicBoolean(false);
+
+  @Override
+  public int defaultTimeoutSeconds() {
+    return 2 * 60;
+  }
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    SharedMiniClusterBase.startMiniCluster();
+  }
+
+  @AfterClass
+  public static void teardown() {
+    SharedMiniClusterBase.stopMiniCluster();
+  }
+
+  @Test
+  public void test() throws Exception {
+    final String tableName = getUniqueNames(1)[0];
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+      c.tableOperations().create(tableName);
+      // majc should take 100 * 5 secs
+      IteratorSetting it = new IteratorSetting(100, SlowIterator.class);
+      SlowIterator.setSleepTime(it, 5000);
+      c.tableOperations().attachIterator(tableName, it,
+          EnumSet.of(IteratorUtil.IteratorScope.majc));
+      try (BatchWriter bw = c.createBatchWriter(tableName)) {
+        for (int i = 0; i < 100; i++) {
+          Mutation m = new Mutation("" + i);
+          m.put("cf", "cq", new Value());
+          bw.addMutation(m);
+        }
+        bw.flush();
+      }
+      // start majc
+      final AtomicReference<Exception> ex = new AtomicReference<>();
+      Thread thread = new Thread(() -> {
+        try {
+          log.info("Compact " + tableName);
+          started.set(true);
+          c.tableOperations().compact(tableName, null, null, true, true);
+          log.info("Finished Compact " + tableName);
+          finished.set(true);
+        } catch (Exception e) {
+          ex.set(e);
+        }
+      });
+      thread.start();
+
+      while (!started.get()) {
+        sleepUninterruptibly(1, TimeUnit.SECONDS);
+      }
+      // delete the table, interrupts the compaction
+      log.info("Delete table: " + tableName);
+      c.tableOperations().delete(tableName);
+
+      while (!finished.get()) {
+        log.info("Wait for compaction to be cancelled");
+        sleepUninterruptibly(5, TimeUnit.SECONDS);
+      }
+      assertTrue(finished.get());
```

Comment:
Shouldn't this be false and ex.get() != null if the compaction was interrupted by the delete?

Reply:
Possibly. I am not sure what will happen with the client yet as we haven't implemented the cancel.